kopf/_core/engines/admission.py

```diff
@@ -126,6 +126,9 @@ async def serve_admission_request(
 
     memo = await memories.recall_memo(raw_body, memobase=memobase, ephemeral=operation=='CREATE')
     body = bodies.Body(raw_body)
+    old = bodies.BodyEssence(old_body) if old_body is not None else None
+    new = bodies.BodyEssence(new_body) if new_body is not None else None
```

Comment:
Thanks for the contribution. Generally, looks great and the feature might be useful. 

The only concern is here, in these lines. The body essence is different from the body — the essence removes all the garbage. The definition of "garbage" is flexible, but mostly it is the system metadata: e.g. generation/resourceVersion/timestamps. I'm not sure how Kubernetes passes the metadata in admission hooks — I simply didn't pay attention to it before. So, I will perform some experiments locally to see how it fits and feels before merging.

_As a side-note (no action needed; for my own remembering later): In the change-detecting handlers, the essence also removes the status stanza and re-adds the explicitly monitored fields. But that was done for the purpose of not (over-)reacting to non-essential changes, i.e. to dummy/technical events or other controllers storing their status fields (or even annotations of other Kopf-based controllers). In the case of admission, the reaction happens anyway, so there is no big win in removing the status/annotations. It can actually be more confusing if a change is admitted, but the diff is empty — so, it should not be made empty. [/the end of the side-note]_

Reply:
Would you prefer this to be the body or the body essence? I am good with either.